rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystem.java

```diff
@@ -306,22 +252,36 @@ else if (start >= 0L && len >= 0L) {
       else {
         // Using similar logic of returning all Blocks as FileSystem.getFileBlockLocations does instead of only returning blocks from start till len
 
-        BlockLocation[] blockLocations = new BlockLocation[(int) Math.ceil((double) file.getLen() / clusterManager.getSplitSize())];
-        int blockNumber = 0;
-        for (long i = 0; i < file.getLen(); i = i + clusterManager.getSplitSize()) {
-          long end = i + clusterManager.getSplitSize();
-          if (end > file.getLen()) {
-            end = file.getLen();
+        try {
+          BlockLocation[] blockLocations = new BlockLocation[(int) Math.ceil((double) file.getLen() / splitSize)];
+          int blockNumber = 0;
+
+          RetryingBookkeeperClient client = new BookKeeperFactory().createBookKeeperClient(conf);
+
+          for (long i = 0; i < file.getLen(); i = i + splitSize) {
+            long end = i + splitSize;
+            if (end > file.getLen()) {
+              end = file.getLen();
+            }
+            String key = file.getPath().toString() + i + end;
+            String hostName = client.getOwnerNodeForPath(key);
```

Comment:
Is there a plan to benchmark this change? Thrift calls in planning phase are bit of a concern given that we see high socket read times once in a while in read phase.